workshops/git_101/02-clone-commit-push-merge.md

```diff
@@ -0,0 +1,161 @@
+
+## Cloning a repository
+
+Cloning is making a local copy of code in a local copy of the remote repository. Most of the time you will be cloning someone elses repository so you can use it or make a change and then push in for a request to have it merged into the upstream codebase. Remember unless you have access to the upstream repository you will need to either request access or fork the code to push into it.
+
+For this exercise fork this repository using the github instructions above then clone it (depending on if you created a PAT(public access token) or a SSH key follow the correct instruction below:
+
+`Note:` You can import into [Gitlab](https://docs.gitlab.com/ee/user/project/import/github.html), [ADO](https://docs.microsoft.com/en-us/azure/devops/repos/git/import-git-repository?view=azure-devops), [Bitbucket](https://support.atlassian.com/bitbucket-cloud/docs/import-a-repository-from-github-or-gitlab/) or git server of choice using their import feature if you do not have access to github (click link or tool for details).
+
+### SSH Clone
+
+```shell
+git clone git@github.com:yourgithubname/HowWeEnable.git
+```
+
+### HTTP Clone
+
+```shell
+git clone https://<pat>@github.com/yourgithubusername/HowWeEnable.git
+```
+
+`NOTE:` Replace `<pat>` with the token you created in the earlier exercise.
+
+Output:
+
+```shell
+Cloning into 'HowWeEnable'...
+remote: Enumerating objects: 283, done.
+remote: Counting objects: 100% (283/283), done.
+remote: Compressing objects: 100% (173/173), done.
+remote: Total 283 (delta 77), reused 276 (delta 74), pack-reused 0
+Receiving objects: 100% (283/283), 81.03 KiB | 1.72 MiB/s, done.
+Resolving deltas: 100% (77/77), done.
+cd HowWeEnable
+tree
+.
+├── LICENSE
+├── README.md
+└── workshops
+    └── how_we_enable
+        ├── 01-Getting_Started.md
+        ├── 02-Logistics.md
+        ├── 03-Strategy.md
+        └── 04-Practice_Scenarios.md
+```
+
+## Branching
+
+It is a best practice to not commit code directly to the main (legacy master) branch. In order to void this we create a feature branch to work from with the `git branch` or `git checkout` command.
+
+To create a branch and check it out (ie move to it as your working branch) you can do this with a single command `git checkout -b <branch-name>`.
+
+```shell
+cd HowWeEnable
+git checkout -b test
+```
+
+Output:
+
+```shell
+Switched to a new branch 'test'
+```
+
+Now any changes you make are in a separate branch from main called test.
+
+## Commiting Code
+
+1. In the HowWeEnable directory create a new file and look at the status of it in git.
+
+    ```shell
+    touch mynewfile.txt
+    git status
+    ```
+
+    Output of git status:
+
+    ```shell
+    Untracked files:
+    (use "git add <file>..." to include in what will be committed)
+        mynewfile.txt
+    ```
+
+2. Now let's git add so the file is tracked by git going forward, and git commit so the new file is ready to be pushed to the remote repository.
```

Comment:
Add code syntax for `git add` and `git commit`

Reply:
```suggestion
2. Now let's add the file so it is tracked by git going forward `git add <filename>, and commit it so that the new file is ready to be pushed to the remote repository `git commit -m 'meaningful commit message'.
```